app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :external_urls, :as => :refers_to, :dependent => :destroy
+  end
+
+  def external_url=(url, user)
```

Comment:
What do you think about defaulting the `user` parameter?

`def external_url=(url, user = User.current_user)`